internal/services/pubkey_service.go

```diff
@@ -99,6 +117,48 @@ func DeletePubkey(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
+	pubkeyResource, err := dao.GetPubkeyResourceDao(r.Context())
+	if err != nil {
+		renderError(w, r, payloads.NewInitializeDAOError(r.Context(), "pubkey resource DAO", err))
+		return
+	}
+
+	resources, err := pubkeyResource.ListByPubkeyId(r.Context(), id)
+	if err != nil {
+		renderError(w, r, payloads.NewDAOError(r.Context(), "delete pubkey", err))
+		return
+	}
+
+	for _, res := range resources {
+		if res.Handle != "" {
+			arn, errArn := sourcesClient.GetArn(r.Context(), res.SourceID)
+			if errArn != nil {
+				if errors.Is(err, sources.ApplicationNotFoundErr) {
+					renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources: application not found", err, 404))
+					return
+				}
+				if errors.Is(err, sources.AuthenticationForSourcesNotFoundErr) {
+					renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources: authentication not found", err, 404))
+					return
+				}
+				renderError(w, r, payloads.ClientError(r.Context(), "Sources", "can't fetch arn from sources", err, 500))
+				return
+			}
+			crd, err := stsClient.AssumeRole(arn)
+			if err != nil {
+				renderError(w, r, payloads.NewAWSError(r.Context(), "assume role sts", err))
+				return
+			}
```

Comment:
We've to assume every resource has been created under different account and later on even under different cloud provider, which makes this a necessity, but we can now `clients.GetCustomerAccount(ctx, arn, region)` we do not have the region, which kinda sucks, but I'd be ok having static region for now, but that means it will be necessary to also keep track of which region the key has been created in.